gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaExtractor.java

```diff
@@ -281,29 +285,36 @@ private void moveToNextPartition() {
           this.highWatermark.get(this.currentPartitionIdx) - this.nextWatermark.get(this.currentPartitionIdx)));
       switchMetricContextToCurrentPartition();
     }
-    this.stopwatch.start();
+
+    if (!allPartitionsFinished()) {
+      this.startFetchEpochTime.put(this.getCurrentPartition(), System.currentTimeMillis());
+    }
   }
 
-  private void updateStatisticsForCurrentPartition() {
-    this.stopwatch.stop();
+  protected void updateStatisticsForCurrentPartition() {
+    long stopFetchEpochTime = System.currentTimeMillis();
+
+    if (!allPartitionsFinished()) {
```

Comment:
I added the check there because updateStatisticsForCurrentPartition is called from the close method, which I think happens after the currentPartitionIdx goes out of bounds, although now that I think about it, it seems like this call from close wouldn't be doing anything useful...